multimap/src/test/java/org/infinispan/multimap/impl/EmbeddedMultimapCacheTest.java

```diff
@@ -0,0 +1,163 @@
+package org.infinispan.multimap.impl;
+
+import static org.infinispan.functional.FunctionalTestUtils.await;
+import static org.testng.AssertJUnit.assertEquals;
+import static org.testng.AssertJUnit.assertFalse;
+import static org.testng.AssertJUnit.assertTrue;
+
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.multimap.EmbeddedMultimapCacheManagerFactory;
+import org.infinispan.multimap.api.MultimapCache;
+import org.infinispan.test.SingleCacheManagerTest;
+import org.infinispan.test.fwk.TestCacheManagerFactory;
+import org.testng.annotations.Test;
+
+/**
+ * Single Multimap Cache Test
+ *
+ * @author Katia Aresti, karesti@redhat.com
+ * @since 9.1
+ */
+@Test(groups = "functional", testName = "multimap.EmbeddedMultimapCacheTest")
+public class EmbeddedMultimapCacheTest extends SingleCacheManagerTest {
+   protected MultimapCache<String, String> multimapCache;
+
+   @Override
+   protected EmbeddedCacheManager createCacheManager() throws Exception {
+      // start a single cache instance
+      ConfigurationBuilder c = getDefaultStandaloneCacheConfig(false);
+      EmbeddedCacheManager cm = TestCacheManagerFactory.createCacheManager(false);
+      multimapCache = EmbeddedMultimapCacheManagerFactory.get("test", cm, c.build());
+      return cm;
+   }
+
+   public void testSupportsDuplicates() {
+      assertFalse(multimapCache.supportsDuplicates());
+   }
+
+   public void testPut() {
+      await(
+            multimapCache.put("names", "julien").thenCompose(r1 ->
```

Comment:
You're chaining it wrong, should be:
```java
 multimapCache.put("names", "julien")
   .thenCompose(r -> multimapCache.put("names", "oihana"))
   .thenCompose(r -> multimapCache.put("names", "julien"))
   .thenCompose(r -> multimapCache.get("names"))
```
That applies to the rest of chains as well.